src/styles/main.scss

```diff
@@ -0,0 +1,75 @@
+$item-bg-color: #eee;
+$calendar-item-size: 100px;
+$calendar-gap: 1px;
+$calendar-item-width: $calendar-item-size + $calendar-gap;
+$week: (
+  'mon': 0,
+  'tue': 1,
+  'wed': 2,
+  'thu': 3,
+  'fri': 4,
+  'sat': 5,
+  'sun': 6,
+);
+
+* {
+  margin: 0;
+  padding: 0;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+body {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: 10px;
+  gap: $calendar-gap;
+  width: calc($calendar-item-width * 7);
+
+  &__item {
+    box-sizing: border-box;
+    background-color: $item-bg-color;
+    height: $calendar-item-size;
+    width: $calendar-item-size;
+    border: 1px solid black;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition-duration: 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+    }
+  }
+
+  @for $dayNumber from 1 through 31 {
+    :nth-child(n + #{$dayNumber})::before {
+      content: '#{$dayNumber}';
+    }
+  }
+
+  @for $maxNumber from 28 to 31 {
+    &--month-length-#{$maxNumber} {
+      .calendar__item:nth-child(n + #{$maxNumber + 1}) {
+        display: none;
```

Comment:
The selector `.calendar__item:nth-child(n + #{$maxNumber + 1})` is used to hide calendar items for days that don't exist in shorter months. However, ensure that this logic is only applied within the context of the `.calendar--month-length-#{$maxNumber}` modifier, or it might hide elements unexpectedly in other contexts.